test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -1228,6 +1228,43 @@ UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wireup_unified, rc, "rc")
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wireup_unified, ud, "ud")
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wireup_unified, rc_dc, "rc,dc")
 
+class select_ep_transport : public test_ucp_wireup {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_RMA, TEST_RMA, "rma");
+    }
+
+    char *get_selected_transport(entity *e)
+    {
+        ucp_worker *worker = e->worker();
+        ucp_lane_index_t lane_index = worker->ep_config[0].key.rma_lanes[0];
+        ucp_rsc_index_t rsc_index = worker->ep_config[0].key.lanes[lane_index].rsc_index;
+        return e->ucph()->tl_rscs[rsc_index].tl_rsc.tl_name;
+    }
+};
+
+UCS_TEST_P(select_ep_transport, fp8_round)
+{
+    modify_config("ADDRESS_VERSION", "v2");
+
+    modify_config("NUM_EPS", "65");
+    entity *e1 = create_entity();
+
+    modify_config("NUM_EPS", "55");
+    entity *e2 = create_entity();
+
+    e1->connect(e2, get_ep_params());
+    char *selected1 = get_selected_transport(e1);
+
+    e2->connect(e1, get_ep_params());
+    char *selected2 = get_selected_transport(e2);
+
+    ASSERT_STREQ(selected1, selected2);
+}
+
+UCP_INSTANTIATE_TEST_CASE_TLS(select_ep_transport, rc_dc, "rc,dc")
```

Comment:
maybe we can run it for all ib transports

Reply:
We only changed common code (nothing for a specific transport).
so we only need to check the general case where you have 2 transports with different priorities for each endpoint.

Reply:
we can do like this - instantiate on multiple transports only:
```
UCP_INSTANTIATE_TEST_CASE_TLS(select_ep_transport, ib, "ib")
UCP_INSTANTIATE_TEST_CASE_TLS(select_ep_transport, ib_shm, "ib,shm")
UCP_INSTANTIATE_TEST_CASE_TLS(select_ep_transport, all, "all")
```